### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by wbaker-figure**

### CODE REVIEW

File: contract-base/src/test/kotlin/io/provenance/scope/contract/TestContract.kt [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
+import io.provenance.scope.contract.proto.HelloWorldExample
 import io.provenance.scope.contract.proto.Specifications
 import io.provenance.scope.contract.proto.TestContractProtos
 import io.provenance.scope.contract.spec.P8eContract
 import io.provenance.scope.contract.spec.P8eScopeSpecification
 
 @Participants([Specifications.PartyType.OWNER])
 @ScopeSpecification(names = ["io.provenance.scope.contract.testcontract"])
-class TestContract: P8eContract() {
+class TestContract(@Record(name = "testRecord") val testRecordValue: List<HelloWorldExample.ExampleName> = mutableListOf(
```
Comment (piercetrey-figure) [inline contract-base/src/test/kotlin/io/provenance/scope/contract/TestContract.kt:16]: why is this record a list of values?

File: sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt [hunk 2]
```diff
@@ -45,18 +49,18 @@ fun createClientDummy(localKeyIndex: Int): Client {
...
-        .setFuncName("record2")
+        .setFuncName("io.provenance.scope.contract.proto.HelloWorldExample\$ExampleName")
         .build()
     val participants = HashMap<Specifications.PartyType, PK.PublicKey>()
     participants[Specifications.PartyType.OWNER] = PK.PublicKey.newBuilder().build()
     val spec = Specifications.ContractSpec.newBuilder()
         .setDefinition(defSpec)
         .addConditionSpecs(conditionSpec)
-        .addFunctionSpecs(Specifications.FunctionSpec.newBuilder().setFuncName("record2").addInputSpecs(defSpec).setInvokerParty(Specifications.PartyType.OWNER))
+        .addFunctionSpecs(Specifications.FunctionSpec.newBuilder().setFuncName("io.provenance.scope.contract.proto.HelloWorldExample\$ExampleName").addInputSpecs(defSpec).setInvokerParty(Specifications.PartyType.OWNER))
```
Comment (piercetrey-figure) [inline sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt:63]: and the resultant record function name

File: sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt [hunk 3]
```diff
@@ -87,13 +91,13 @@ fun createExistingScope(): ScopeResponse.Builder {
     val scopeRecord = Record.newBuilder()
         .addInputs(
             RecordInput.newBuilder()
-            .setName("record2")
-            .setHash("1234567890")
+            .setName("io.provenance.scope.contract.proto.HelloWorldExample\$ExampleName")
```
Comment (piercetrey-figure) [inline sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt:94]: and this 😄

File: sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt [hunk 4]
```diff
@@ -87,13 +91,13 @@ fun createExistingScope(): ScopeResponse.Builder {
...
+            .setHash("M8PWxG2TFfO0YzL3sDW/l9kX325y+3v+5liGcjZoi4Q=")
             .setStatus(RecordInputStatus.RECORD_INPUT_STATUS_PROPOSED))
-        .addOutputs(RecordOutput.newBuilder().setHash("234567834567").build())
-        .setProcess(Process.newBuilder().setName("io.provenance.scope.contract.proto.Contracts\$Record").build())
+        .addOutputs(RecordOutput.newBuilder().setHash("M8PWxG2TFfO0YzL3sDW/l9kX325y+3v+5liGcjZoi4Q=").build())
+        .setProcess(Process.newBuilder().setName("io.provenance.scope.contract.proto.HelloWorldExample\$ExampleName").build())
         .setSessionId(MetadataAddress.forSession(scopeUuid, sessionUUID).bytes.toByteString())
-        .setName("record2")
+        .setName("io.provenance.scope.contract.proto.HelloWorldExample\$ExampleName")
```
Comment (piercetrey-figure) [inline sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt:100]: and this one, was this just a find/replace that did this?

File: sdk/src/test/kotlin/io/provenance/scope/sdk/AffiliateRepositoryTest.kt [hunk 5]
```diff
@@ -10,7 +10,7 @@ import io.provenance.scope.sdk.AffiliateRepository
 import java.security.PublicKey
 
 class AffiliateRepositoryTest : WordSpec() {
-    lateinit var affiliateRepository: AffiliateRepository
+    lateinit var affiliateRepository: io.provenance.scope.sdk.AffiliateRepository
```
Comment (piercetrey-figure) [inline sdk/src/test/kotlin/io/provenance/scope/sdk/AffiliateRepositoryTest.kt:13]: for some reason the package statement for this file seems weird (might have been my bad), seems like it should just be `io.provenance.scope.sdk` given the directory structure

File: sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt [hunk 6]
```diff
@@ -119,3 +123,7 @@ fun createExistingScope(): ScopeResponse.Builder {
     return scopeResponseBuilder
 
 }
+
+data class HelloWorldData(@AnRecord(name = "io.provenance.scope.contract.proto.HelloWorldExample\$ExampleName") val name: HelloWorldExample.ExampleName) {}
```
[the remark this answers is not in the document]
Comment (wbaker-figure) [reply sdk/src/test/kotlin/io/provenance/scope/sdk/SDKTestHelper.kt:127]: Hahaha Yeah, I think I did a find and replace and ran to see if it worked and it did. I can look into switching these back haha

File: sdk/src/test/kotlin/io/provenance/scope/sdk/AffiliateRepositoryTest.kt (hunk 5 above)
Comment (wbaker-figure) [reply sdk/src/test/kotlin/io/provenance/scope/sdk/AffiliateRepositoryTest.kt:13]: Will check that quickly!

File: sdk/src/test/kotlin/io/provenance/scope/sdk/ClientTest.kt [hunk 7]
```diff
@@ -28,18 +41,144 @@ class ClientTest : WordSpec() {
...
+                )
+            )
+
+            client.inner.affiliateRepository.addAffiliate(
+                signingPublicKey = signingKeyPair.public,
+                encryptionPublicKey = encryptionKeyPair.public
+            )
+
+            shouldNotThrowAny {
+                client.hydrate(HelloWorldData::class.java, scope.build())
```
Comment (piercetrey-figure) [inline sdk/src/test/kotlin/io/provenance/scope/sdk/ClientTest.kt:119]: probably would be good to assert some things off of the result of hydration (i.e. that HelloWorldData.name contains the values that CachedOsClient is mocked to return)

Comment (wbaker-figure) [reply sdk/src/test/kotlin/io/provenance/scope/sdk/ClientTest.kt:119]: Will do!

File: contract-base/src/test/kotlin/io/provenance/scope/contract/TestContract.kt (hunk 1 above)
Comment (piercetrey-figure) [reply contract-base/src/test/kotlin/io/provenance/scope/contract/TestContract.kt:16]: oh interesting, I didn't know we supported that, cool

**Comment by piercetrey-figure** [comment]
@wbaker-figure looks like the tests might be failing due to actual grpc communication attempts with Object Store in the `io.provenance.scope.sdk.ClientTest.execute and return a signed result` test, might need to mock the CachedOsClient

**Pull request merged**